File: app/src/main/java/com/panwar2001/pdf2txt/TextPreview.java
package com.panwar2001.pdf2txt;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.FrameLayout;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.ads.AdRequest;
import com.google.android.material.snackbar.Snackbar;
import com.panwar2001.pdf2txt.databinding.ActivityTextpreviewBinding;

public class TextPreview extends AppCompatActivity {
    ActivityTextpreviewBinding binding;
    String unchangedText;
    @Override
    protected void onCreate(@Nullable  Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding =ActivityTextpreviewBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        setSupportActionBar(binding.toolbar.toolbar);
        ActionBar actionBar=getSupportActionBar();
        if(actionBar!=null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            // assign text to EditText
            Intent intent = getIntent();
            unchangedText = intent.getStringExtra("text");
            binding.editText.setText(unchangedText);
        }
        //ads
        AdRequest adRequest = new AdRequest.Builder().build();
        binding.adView.loadAd(adRequest);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu){
        MenuInflater inflater= getMenuInflater();
        inflater.inflate(R.menu.menu,menu);
        return true;
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item){
        int id=item.getItemId();
        if(id==android.R.id.home){
            finish();
            return true;
        }else if(id==R.id.copy) {
            copyToClipBoard();
            return true;
        }else if(id==R.id.undo_changes){
            undoChanges();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
    public void copyToClipBoard(){
        String text= String.valueOf(binding.editText.getText());
        ClipboardManager clipboard=(ClipboardManager)getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clip= ClipData.newPlainText("text",text);
        clipboard.setPrimaryClip(clip);
        SnackBarTop("Copied To Clipboard!");
    }
    public void undoChanges(){
        binding.editText.setText(unchangedText);
        SnackBarTop("Reverted Changes!");
    }
    public void  SnackBarTop(String text){
        Snackbar snack=Snackbar.make(binding.getRoot(),text,Snackbar.ANIMATION_MODE_FADE);
        View view=snack.getView();
        snack.setDuration(300);
        FrameLayout.LayoutParams params=(FrameLayout.LayoutParams) view.getLayoutParams();
        params.gravity= Gravity.TOP;
        view.setLayoutParams(params);
        snack.show();
    }
}

File: app/src/main/java/com/panwar2001/pdf2txt/TextPreview.kt
package com.panwar2001.pdf2txt

import android.content.ClipData
import android.content.ClipboardManager
import android.os.Bundle
import android.view.Gravity
import android.view.Menu
import android.view.MenuItem
import android.widget.FrameLayout
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.ads.AdRequest
import com.google.android.material.snackbar.Snackbar
import com.panwar2001.pdf2txt.databinding.ActivityTextpreviewBinding

class TextPreview : AppCompatActivity() {
    private var binding: ActivityTextpreviewBinding? = null
    private var unchangedText: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTextpreviewBinding.inflate(
            layoutInflater
        )
        setContentView(binding!!.root)
        setSupportActionBar(binding!!.toolbar.toolbar)
        val actionBar:ActionBar?= supportActionBar
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true)
            // assign text to EditText
            val intent = intent
            unchangedText = intent.getStringExtra("text")
            binding!!.editText.setText(unchangedText)
        }
        //ads
        val adRequest = AdRequest.Builder().build()
        binding!!.adView.loadAd(adRequest)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when(item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
            R.id.copy -> {
                copyToClipBoard()
                return true
            }
            R.id.undo_changes -> {
                undoChanges()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

   private fun copyToClipBoard() {
        val text = binding!!.editText.getText().toString()
        val clipboard = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
        val clip = ClipData.newPlainText("text", text)
        clipboard.setPrimaryClip(clip)
        snackBarTop("Copied To Clipboard!")
    }

     private fun undoChanges() {
        binding!!.editText.setText(unchangedText)
        snackBarTop("Reverted Changes!")
    }

    private fun snackBarTop(text: String?) {
        val snack = Snackbar.make(binding!!.root, text!!, Snackbar.ANIMATION_MODE_FADE)
        val view = snack.view
        snack.setDuration(300)
        val params = view.layoutParams as FrameLayout.LayoutParams
        params.gravity = Gravity.TOP
        view.setLayoutParams(params)
        snack.show()
    }
}
